### PR TITLE
user crud

### PR DESCRIPTION
**Pull request by oarozu**
Crear User controller con las operaciones CRUD de User

https://trello.com/c/BuaB2yAe/21-crear-user-controller-con-las-operaciones-crud-de-user

### CODE REVIEW

File: src/main/java/wolox/training/controllers/UserController.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+import wolox.training.services.UserService;
+
+import java.security.Principal;
+
+@RestController
+@RequestMapping("/api/users")
+public class UserController {
+
+    @Autowired
+    private BookService bookRepository;
```
Comment (cramallo) [inline src/main/java/wolox/training/controllers/UserController.java]: remove the repository from here, you should use the repository in the service

File: src/main/java/wolox/training/controllers/UserController.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    public User addBookToUser(@PathVariable Long user_id, @PathVariable Long book_id) {
+        User user = userService.findById(user_id);
+        Book book = bookRepository.findById(book_id);
+        user.addBook(book);
+        return userService.save(user);
+    }
+
+    @DeleteMapping("/{user_id}/books/{book_id}")
+    public User deleteBookToUser(@PathVariable Long user_id, @PathVariable Long book_id) {
+        User user = userService.findById(user_id);
```
Comment (cramallo) [inline src/main/java/wolox/training/controllers/UserController.java]: You can do all this in the service

File: src/main/java/wolox/training/services/UserService.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import wolox.training.exceptions.BookNotFoundException;
+import wolox.training.models.Book;
+import wolox.training.models.User;
+import wolox.training.repositories.UserRepository;
+
+import java.util.List;
+import java.util.Optional;
+
+public class UserService {
```
Comment (cramallo) [inline src/main/java/wolox/training/services/UserService.java:14]: use @Service annotation

**Review by cramallo — CHANGES_REQUESTED** [review]

**Review by cramallo — COMMENTED** [review]
Remember the uses of "final" in variables that won't never change

File: src/main/java/wolox/training/controllers/UserController.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+import java.security.Principal;
+
+@RestController
+@RequestMapping("/api/users")
+public class UserController {
+
+    @Autowired
+    private UserService userService;
+    @Autowired
+    private BookService bookService;
```
Comment (cramallo) [inline src/main/java/wolox/training/controllers/UserController.java]: you never use this service here, you should remove this

**Review by cramallo — APPROVED** [review]

**Review by facucompe — APPROVED** [review]

**Pull request merged**